src/unknown_00BC20.c

```diff
@@ -600,7 +600,72 @@ void func_8000E1EC(Object *object, s32 arg1) {
     gObjectCount = 0;
 }
 
+#ifdef NON_MATCHING
+typedef struct LevelObjectEntry8000E2B4 {
+    LevelObjectEntryCommon common;
+    s16 unk8;
+    s16 unkA;
+    s16 unkC;
+    s16 unkE;
+} LevelObjectEntry8000E2B4;
+
+void func_8000E2B4(void) {
+    LevelObjectEntry8000E2B4 sp2C;
+    Settings *settings;
+    Object *player;
+    Object_64 *player_64;
+    s16 object_id;
+
+    if (D_8011AD44 == 0) {
+        return;
+    }
+    D_8011AD44 = (s8) (D_8011AD44 - 1);
+    if (D_8011AD44 != 0) {
+        return;
+    }
+    settings = get_settings();
+    sp2C.unkE = 0;
+    sp2C.common.size = 0x10;
+    if (D_8011AD45 < 5) {
+        object_id = ((s16*)D_800DC7A8)[settings->racers[0].character + D_8011AD45 * 10];
+    } else {
+        object_id = D_800DC7B8[D_8011AD45 + 37];
+    }
+    func_8006DB14(D_8011AD45);
+    sp2C.common.size = sp2C.common.size | ((s32) (object_id & 0x100) >> 1);
+    sp2C.unkA = 0;
+    sp2C.unk8 = 0;
+    sp2C.common.object_id = (s8) object_id;
+    sp2C.common.x = D_8011AD46;
+    sp2C.common.y = D_8011AD48;
+    sp2C.common.z = D_8011AD4A;
+    sp2C.unkC = D_8011AD4C;
+    func_800521B8(1);
+    player = func_8000EA54(&sp2C, 0x11);
+    gObjectCount = 1;
+    (*gObjectStructArrayPtr)[0] = (s32) player;
+    *D_8011AEEC = (s32) player;
+    *D_8011AEE8 = (s32) player;
+    player_64 = player->unk64;
+    player_64->unk1D6 = (s8) D_8011AD45;
```

Comment:
unk1D6 is undefined here. Check to see if maybe you forgot to update the Object_64 struct in this PR?

Reply:
Done in https://github.com/DavidSM64/Diddy-Kong-Racing/pull/178/commits/e93a5db1ecfa2e29a151e673a3f295464e067349. Thanks.